From 9faeef04b2151c87d2dae6c2b0af9c1d826cc5c2 Mon Sep 17 00:00:00 2001
From: Masami Ichikawa <masami.ichikawa@miraclelinux.com>
Date: Tue, 16 Jul 2019 10:13:34 +0900
Subject: [PATCH] Revert python version patch

Poky uses python2 and python3 as python command name, so keeping poky'
specification, it needs to revert
0003-python-Auto-check-for-all-installed-python-versions.patch.

Signed-off-by: Masami Ichikawa <masami.ichikawa@miraclelinux.com>
---
 configure.ac | 44 ++++++++++++++++++++++++++++++++++----------
 m4/python.m4 |  9 +++------
 2 files changed, 37 insertions(+), 16 deletions(-)

diff --git a/configure.ac b/configure.ac
index d23915c..3293245 100644
--- a/configure.ac
+++ b/configure.ac
@@ -184,7 +184,7 @@ have_w64_system=no
 have_macos_system=no
 build_w32_glib=no
 build_w32_qt=no
-available_languages="cl cpp python qt"
+available_languages="cl cpp python python2 python3 qt"
 default_languages="cl cpp python qt"
 case "${host}" in
     x86_64-*mingw32*)
@@ -379,9 +379,13 @@ fi
 AC_SUBST(HAVE_DOT)
 
 # Python bindings.
+LIST_MEMBER("python2", $enabled_languages)
+found_py2=$found
+LIST_MEMBER("python3", $enabled_languages)
+found_py3=$found
 LIST_MEMBER("python", $enabled_languages)
 found_py=$found
-if test "$found_py" = "1"; then
+if test "$found_py" = "1" -o "$found_py2" = "1" -o "$found_py3" = "1"; then
     AX_PKG_SWIG
     if test -z "$SWIG"; then
         if test "$explicit_languages" = "1"; then
@@ -393,15 +397,36 @@ if test "$found_py" = "1"; then
             enabled_languages=$(echo $enabled_languages | sed 's/python//')
         fi
     else
-        # Reset the version collecting vars.
+        # Reset all the stuff, just to be sure.
 	PYTHONS=
 	PYTHON_VERSIONS=
+	unset PYTHON
+	unset PYTHON_VERSION
+	unset PYTHON_CPPFLAGS
+	unset PYTHON_LDFLAGS
+	unset PYTHON_SITE_PKG
+	unset PYTHON_EXTRA_LIBS
+	unset PYTHON_EXTRA_LDFLAGS
+	unset ac_cv_path_PYTHON
+	unset am_cv_pathless_PYTHON
+	unset am_cv_python_version
+	unset am_cv_python_platform
+	unset am_cv_python_pythondir
+	unset am_cv_python_pyexecdir
+
+	if test "$found_py" = "1" -o "$found_py2" = "1"; then
+	    AM_PATH_PYTHON([2.7], [
+		AX_PYTHON_DEVEL
+		if test "$PYTHON_VERSION"; then
+			PYTHONS="$(echo $PYTHONS $PYTHON)"
+			PYTHON_VERSIONS="$(echo $PYTHON_VERSIONS $PYTHON_VERSION)"
+		fi
+	    ], :)
+	fi
 
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
-	  # Reset everything, so that we can look for another Python.
-          m4_foreach([mym4pythonver],
-                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[all]],
-           [unset PYTHON
+	    # Reset everything, so that we can look for another Python.
+	    unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
 	    unset PYTHON_LDFLAGS
@@ -414,14 +439,13 @@ if test "$found_py" = "1"; then
 	    unset am_cv_python_platform
 	    unset am_cv_python_pythondir
 	    unset am_cv_python_pyexecdir
-	    AM_PATH_PYTHON(mym4pythonver, [
+	    AM_PATH_PYTHON([3.4], [
 		AX_PYTHON_DEVEL
 		if test "$PYTHON_VERSION"; then
 			PYTHONS="$(echo $PYTHONS $PYTHON)"
 			PYTHON_VERSIONS="$(echo $PYTHON_VERSIONS $PYTHON_VERSION)"
 		fi
-	    ], :, m4_if([mym4pythonver],[all],[],[python]mym4pythonver))
-           ])
+	    ], :)
 	fi
 
 	# Recover some values lost in the second attempt to find Python.
diff --git a/m4/python.m4 b/m4/python.m4
index 7adcdea..7097db2 100644
--- a/m4/python.m4
+++ b/m4/python.m4
@@ -2,8 +2,7 @@
 ## Python file handling
 ## From Andrew Dalke
 ## Updated by James Henstridge
-## Upated by Werner Koch 2018-10-17
-## ---------------------------------
+## ------------------------
 # Copyright (C) 1999-2017 Free Software Foundation, Inc.
 #
 # This file is free software; the Free Software Foundation
@@ -11,8 +10,7 @@
 # with or without modifications, as long as this notice is preserved.
 
 
-# AM_PATH_PYTHON([MINIMUM-VERSION], [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND]
-#                [INTERPRETER-LIST])
+# AM_PATH_PYTHON([MINIMUM-VERSION], [ACTION-IF-FOUND], [ACTION-IF-NOT-FOUND])
 # ---------------------------------------------------------------------------
 # Adds support for distributing Python modules and packages.  To
 # install modules, copy them to $(pythondir), using the python_PYTHON
@@ -69,8 +67,7 @@ AC_DEFUN([AM_PATH_PYTHON],
       # VERSION.
       AC_CACHE_CHECK([for a Python interpreter with version >= $1],
 	[am_cv_pathless_PYTHON],[
-	for am_cv_pathless_PYTHON dnl
-          in m4_if([$4],[],[_AM_PYTHON_INTERPRETER_LIST],[$4]) none; do
+	for am_cv_pathless_PYTHON in _AM_PYTHON_INTERPRETER_LIST none; do
 	  test "$am_cv_pathless_PYTHON" = none && break
 	  AM_PYTHON_CHECK_VERSION([$am_cv_pathless_PYTHON], [$1], [break])
 	done])
-- 
2.20.1

